tests/expectations/test_self_check.py

```diff
@@ -0,0 +1,448 @@
+        include_column_name = runtime_configuration.get("include_column_name", True)
+        include_column_name = (
+            include_column_name if include_column_name is not None else True
+        )
+        styling = runtime_configuration.get("styling")
+        params = substitute_none_for_missing(
+            configuration.kwargs,
+            ["column", "regex", "mostly", "row_condition", "condition_parser"],
+        )
+
+        template_str = "values must be equal to 3"
+        if params["mostly"] is not None:
+            params["mostly_pct"] = num_to_str(
+                params["mostly"] * 100, precision=15, no_scientific=True
+            )
+            # params["mostly_pct"] = "{:.14f}".format(params["mostly"]*100).rstrip("0").rstrip(".")
+            template_str += ", at least $mostly_pct % of the time."
+        else:
+            template_str += "."
+
+        if include_column_name:
+            template_str = "$column " + template_str
+
+        if params["row_condition"] is not None:
+            (
+                conditional_template_str,
+                conditional_params,
+            ) = parse_row_condition_string_pandas_engine(params["row_condition"])
+            template_str = conditional_template_str + ", then " + template_str
+            params.update(conditional_params)
+
+        return [
+            RenderedStringTemplateContent(
+                **{
+                    "content_block_type": "string_template",
+                    "string_template": {
+                        "template": template_str,
+                        "params": params,
+                        "styling": styling,
+                    },
+                }
+            )
+        ]
+
+
+def test_expectation_self_check():
+
+    my_expectation = ExpectColumnValuesToEqualThree(
+        configuration=ExpectationConfiguration(
+            **{
+                "expectation_type": "expect_column_values_to_equal_three",
+                "kwargs": {"column": "threes"},
+            }
+        )
+    )
+    report_object = my_expectation.self_check()
+    print(json.dumps(report_object, indent=2))
+
+    assert report_object == {
+        "description": {
+            "camel_name": "ExpectColumnValuesToEqualThree",
+            "snake_name": "expect_column_values_to_equal_three",
+            "short_description": "",
+            "docstring": "",
+        },
+        "renderers": {},
+        "examples": [],
+        "metrics": [],
+        "execution_engines": {},
+        "library_metadata": {},
+    }
+
+
+def test_all_expectation_self_checks():
+    library_json = {}
+
+    for expectation_name, expectation in _registered_expectations.items():
+        report_object = expectation().self_check()
+        library_json[expectation_name] = report_object
+        print(report_object["metrics"])
+
+    # with open('output/expectation_library.json', 'w') as f_:
+    #     f_.write(json.dumps(library_json, indent=2))
+
```

Comment:
This should be removed from the test suite and instead moved to a utility module inside docs.

Reply:
Done.